xdem/coreg/base.py

```diff
@@ -1351,6 +1351,140 @@ def apply(
         else:
             return applied_elev, out_transform
 
+    @overload
+    def fit_and_apply(
```

Comment:
Simple ! 
I think 99% of the users will only need to use that function. Maybe we could just add a note in the other functions `fit` and `apply` that this one should be used in priority. The only reasons I can see to split the two is 1) if making changes to the coregistration or 2) to run the same coregistration to several DEMs. Do you see any other reason? 
It would almost make sense to hide the two step functions...

Reply:
I added an issue to make sure we update the examples in the doc to encourage using this function: https://github.com/GlacioHack/xdem/issues/532